src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -37,6 +37,43 @@ export const configureClient = async (
   }
 };
 
+export const configureTestClient = async (
+  { savedObjects, cryptography }: DataSourceClientParams,
+  dataSource: DataSourceAttributes,
+  openSearchClientPoolSetup: OpenSearchClientPoolSetup,
+  config: DataSourcePluginConfigType,
+  logger: Logger
+): Promise<Client> => {
+  try {
+    const {
+      id,
+      auth: { type, credentials },
+    } = dataSource;
+    let requireDecryption = false;
+
+    const rootClient = getRootClient(dataSource, config, openSearchClientPoolSetup);
```

Comment:
I'd still suggest to not use any of `getRootClient`, `getQueryClient` for test clients. It doesn't make sense to add a test client to the connection pool. I suggest just use the most basic way below to build client. This way you don't need to refactor any existing logic.
```
const client = new Client({auth, endpoint, etc})
```